### PR TITLE
fix: type correction for `loading`, `loadingIndicator`, `transpile`

### PR DESCRIPTION
**Pull request by wattanx**
### 🔗 Linked issue
Fixes: https://github.com/nuxt/bridge/issues/1019
Fixes: https://github.com/nuxt/bridge/issues/1016

### ❓ Type of change

- [ ] 📖 Documentation (updates to the documentation or readme)
- [x] 🐞 Bug fix (a non-breaking change that fixes an issue)
- [ ] 👌 Enhancement (improving an existing functionality like performance)
- [ ] ✨ New feature (a non-breaking change that adds functionality)
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)

### 📚 Description
I will correct the types for `loading`, `loadingIndicator`, `transpile` in the nuxt.config.
I would like to add type tests in a separate PR.

### 📝 Checklist

- [ ] I have linked an issue or discussion.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: packages/bridge-schema/src/config/build.ts [hunk 1]
```diff
@@ -517,6 +517,21 @@ export default defineUntypedSchema({
...
+     * dependency's file name.
+     *
+     * You can also use a function to conditionally transpile. The function will receive an object ({ isDev, isServer, isClient, isModern, isLegacy }).
+     * @example
+     * ```js
+     transpile: [({ isLegacy }) => isLegacy && 'ky']
+     * ```
+     * @type {Array<string | RegExp | ((ctx: { isClient?: boolean; isServer?: boolean; isDev: boolean }) => string | RegExp | false)>}
+     */
+    transpile: {
```
Comment (yanniznik) [inline packages/bridge-schema/src/config/build.ts:533]: Nice <3

File: packages/bridge-schema/src/config/app.ts [hunk 2]
```diff
@@ -254,6 +246,7 @@ export default defineUntypedSchema({
    * configuration. The name can refer to an indicator from [SpinKit](https://tobiasahlin.com/spinkit/)
    * or a path to an HTML template of the indicator source code (in this case, all the
    * other options will be passed to the template).
+   * @type {typeof import('../src/types/loading').NuxtOptionsLoadingIndicator | false}
```
Comment (wJoenn) [inline packages/bridge-schema/src/config/app.ts:249]: It's still missing the type `false` isn't it ?

Comment (wattanx) [reply packages/bridge-schema/src/config/app.ts:249]: I have confirmed that setting `false` does not cause an error, what do you think is the problem?

**Review by danielroe — APPROVED** [review]

File: packages/bridge-schema/src/config/app.ts (hunk 2 above)
Comment (wJoenn) [reply packages/bridge-schema/src/config/app.ts:249]: Here's the current issue I have
image

And here's the attribute in the `node_modules/@nuxt/bridge-schema/dist/index.d.ts` `ConfigSchema` interface
```ts
 /**
  * Used to set the default properties of the page transitions.
  * 
  * You can either pass a string (the transition name) or an object with properties to bind to the `<Transition>` component that will wrap your pages.
  * 
  * 
  * @see [vue@2 documentation](https://v2.vuejs.org/v2/guide/transitions.html)
  * 
  * @see [vue@3 documentation](https://vuejs.org/guide/built-ins/transition-group.html#enter-leave-transitions)
 */
 pageTransition: {
  [key: string]: any
 },
 ```

**Comment by wJoenn** [comment]
@danielroe I just discovered another type mismatch (currently installing bridge on a old Nuxt 2 app).
`app.meta.script` raises an error when a `body: true` attribute is passed to a script
```ts
app: {
  head: {
    script: [
      { src: `https://polyfill.io/v3/polyfill.min.js?features=${POLYFILL_FEATURES}`, body: true }
    ]
  }
}
```
image

But I can see in Unhead's [documentation](https://unhead.unjs.io/usage/guides/positions#tagposition) that

> Providing body: true is the same as tagPosition: 'bodyClose'

`tagPosition: 'bodyClose'` does not raise an error. `body: true` does.

Shall I open a new issue for this one ?

**Pull request merged**